docs/reference/adaptors.rst

```diff
@@ -383,6 +383,40 @@ Adaptors
         * :func:`flux::scan`
         * :func:`flux::fold_first`
 
+``set_union``
+^^^^^^^^^^^^^
+
+..  function::
+    template <sequence Seq1, sequence Seq2, typename Cmp = std::ranges::less> \
+        requires compatible<Seq1, Seq2> and \
+                 strict_weak_order_for<Cmp, Seq1> and \
+                 strict_weak_order_for<Cmp, Seq2> \
+    auto set_union(Seq1&& seq1, Seq2&& seq2, Cmp cmp = {}) -> sequence auto;
+
+    Returns a sequence adaptor which yields the set union of the two input sequences :var:`seq1` and :var:`seq2`, ordered by the given comparison function :var:`cmp`.
+
+    This function assumes that both :var:`seq1` and :var:`seq2` are sorted in ascending order with respect to the comparison function :var:`cmp`. If the input sequences are not sorted, the behavior is undefined.
+
+    The resulting sequence is stateful, and its iteration is based on the two input sequences, so it requires at most one pass through each input sequence. When the resulting sequence is iterated, it will output the elements from the two input sequences in order according to :var:`cmp`, but without duplicates.
```

Comment:
There's no need to say that the resulting sequence is stateful -- it is no different to other sequence adaptors in that regard.

Also, it's not quite correct to say that it's "without duplicates" -- as implemented, your `set_union` has the same behaviour as `std::set_union` in that if seq1 has `n` copies of an element and seq2 has `m` copies, the resulting sequence will have `max(n, m)` copies.